### PR TITLE
Tearout from favourites tab

### PR DESCRIPTION
**Pull request by owennw**
Dragging a card outside of the favourites tab and releasing will now create a new window. Note that tearing in still applies to the whole window. Perhaps it should be changed for symmetry too? #435

### CODE REVIEW

File: src/sidebars/favourites/tearout.js [hunk 1]
```diff
@@ -167,11 +167,9 @@
                                                 dragService = null;
                                             } else {
                                                 // Create new window instance
-                                                var mainApplicationWindowPosition = geometryService.getWindowPosition(window);
-
                                                 var compact = store.isCompact();
                                                 windowService.createMainWindow(null, compact, (newWindow) => {
-                                                    newWindow.resizeTo(mainApplicationWindowPosition.width, mainApplicationWindowPosition.height, 'top-left');
+                                                    newWindow.resizeTo(window.outerWidth, window.outerHeight, 'top-left');
```
Comment (WPFerg) [inline src/sidebars/favourites/tearout.js]: Does this resize the new window to the size of the window you dragged out from?

File: src/sidebars/favourites/tearout.js [hunk 2]
```diff
@@ -191,13 +189,11 @@
...
 
-                            function reorderFavourites(tearoutRectangle) {
+                            function reorderFavourites() {
                                 var hoverTargets = hoverService.get();
 
                                 for (var i = 0, max = hoverTargets.length; i < max; i++) {
-                                    var dropTargetRectangle = geometryService.rectangle(
-                                        geometryService.elementScreenPosition(geometryService.getWindowPosition(window), hoverTargets[i].hoverArea)),
-                                        overDropTarget = tearoutRectangle.intersects(dropTargetRectangle);
+                                    var overDropTarget = geometryService.elementIntersect(tearoutWindow, window, hoverTargets[i].hoverArea);
```
Comment (WPFerg) [inline src/sidebars/favourites/tearout.js]: This and https://github.com/ScottLogic/bitflux-openfin/pull/440/files#diff-3709fe15a75397a1b30d1833b8c85403R211 are quite long lines, could you split them up?

File: src/sidebars/favourites/tearout.js (hunk 1 above)
Comment (owennw) [inline src/sidebars/favourites/tearout.js]: Yes, just unravelling the pointless call to the geometry service, the code is identical though.

File: src/sidebars/favourites/tearout.js (hunk 2 above)
Comment (owennw) [inline src/sidebars/favourites/tearout.js]: They're both within the line character limit =/

**Comment by WPFerg** [comment]
I'll hold off from merging this for now until we decide whether it's worth applying this to tearing in.

**Comment by owennw** [comment]
How shall we decide?

**Comment by WPFerg** [comment]
I'm not sure, I think it's probably best if it's symmetrical. How difficult would it be to do?

**Comment by owennw** [comment]
I'll merge this as I think it provides useful functionality on its own. With it merged we can re-test and see whether implementing the symmetrical idea is worthwhile.

WRT the difficulty, it's a matter of finding the favourites tab element in the global window service when comparing intersections of windows. Not sure how cumbersome that would be.

**Pull request merged**